### ISSUE TITLE
Installing add-ons on Windows updates numpy (and makes Orange useless)

### ISSUE THREAD

**Issue by echavezangel** [biolab/orange3#5765]
I am getting a the following problem with data set files ("tab" files) after installing some adds 
Read error:
'''Cannot parse dataset C:/Program Files/Orange/Lib/site-packages/Orange/datasets/housing.tab: Cannot cast ufunc '__contains__ (vectorized)' output from dtype('O') to dtype('bool') with casting rule 'same_kind' '''

I don't know exactly which is the add causing this problem.
This appears after the installation of some adds (network, time series or any except spectroscopy). 
I updated the orange version 3.30.2 -> 3.31.0 but the problem didn't disappear. The only way to read again "tab" files is uninstalling orange and installing again.  

- Operating system: Win 10 64 bits
- Orange version: 3.30.2 and 3.31.0
- How you installed Orange: *.exe

**Comment by markotoplak** [comment]
On Dec 31 a new version of numpy was released which seemed to get updated during your add-on installation process. The new version of numpy introduced some changes which are incompatible with the current Orange code. We plan to update Orange to support newer numpy versions shortly (the fix was already committed, see #5760).

To confirm that this is the problem could you please open the Orange command prompt and paste the outputs of `conda list` and `conda --version` here?

You can solve your problem by installing the previous numpy version manually. Open the Orange command prompt and run `conda install numpy=1.21`. Please report if that solved your problem. Do not forget to run Orange command prompt as admin.

Does anyone have any idea why would numpy get upgraded when installing add-ons? All but spectroscopy are installed with `conda`, so It is likely something `conda install` related. If numpy gets upgraded for at every add-on install (I suspect none has 1.22 as an explicit dependency) we have a problem...

**Comment by echavezangel** [comment]
Thank you for the answer the installation of previous numpy worked well.

here the full Conda list:

_py-xgboost-mutex         2.0                       cpu_0    <unknown>
anyio                     3.4.0            py38haa244fe_0    <unknown>
anyqt                     0.0.13             pyh6c4a22f_0    <unknown>
asteval                   0.9.25                   pypi_0    pypi
backcall                  0.2.0              pyh9f0ad1d_0    <unknown>
backports                 1.0                        py_2    <unknown>
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    <unknown>
baycomp                   1.0.2                      py_1    <unknown>
blas                      2.112                  openblas    <unknown>
blas-devel                3.9.0           12_win64_openblas    <unknown>
bottleneck                1.3.2            py38h6f4d8f0_5    <unknown>
brotli                    1.0.9                h8ffe710_6    <unknown>
brotli-bin                1.0.9                h8ffe710_6    <unknown>
brotlipy                  0.7.0           py38h294d835_1003    <unknown>
ca-certificates           2021.10.8            h5b45459_0    <unknown>
cachecontrol              0.12.10            pyhd8ed1ab_0    <unknown>
catboost                  1.0.3            py38haa244fe_1    <unknown>
certifi                   2021.10.8        py38haa244fe_1    <unknown>
cffi                      1.15.0           py38hd8c33c5_0    <unknown>
chardet                   4.0.0            py38haa244fe_2    <unknown>
charset-normalizer        2.0.8              pyhd8ed1ab_0    <unknown>
colorama                  0.4.4              pyh9f0ad1d_0    <unknown>
colorcet                  3.0.0                    pypi_0    pypi
commonmark                0.9.1                      py_0    <unknown>
cryptography              36.0.0           py38hb7941b4_0    <unknown>
cycler                    0.11.0             pyhd8ed1ab_0    <unknown>
debugpy                   1.5.1            py38h885f38d_0    <unknown>
decorator                 5.1.0              pyhd8ed1ab_0    <unknown>
dictdiffer                0.9.0              pyhd8ed1ab_0    <unknown>
docutils                  0.18.1           py38haa244fe_0    <unknown>
entrypoints               0.3             pyhd8ed1ab_1003    <unknown>
et_xmlfile                1.0.1                   py_1001    <unknown>
extranormal3              0.0.3                    pypi_0    pypi
fonttools                 4.28.2           py38h294d835_0    <unknown>
freetype                  2.10.4               h546665d_1    <unknown>
future                    0.18.2           py38haa244fe_4    <unknown>
h11                       0.12.0             pyhd8ed1ab_0    <unknown>
h2                        4.1.0            py38haa244fe_0    <unknown>
h5py                      3.6.0                    pypi_0    pypi
hpack                     4.0.0              pyh9f0ad1d_0    <unknown>
httpcore                  0.13.7             pyhd8ed1ab_0    <unknown>
httpx                     0.19.0           py38haa244fe_0    <unknown>
hyperframe                6.0.1              pyhd8ed1ab_0    <unknown>
icu                       68.2                 h0e60522_0    <unknown>
idna                      3.1                pyhd3deb0d_0    <unknown>
importlib-metadata        4.8.2            py38haa244fe_0    <unknown>
importlib_metadata        4.8.2                hd8ed1ab_0    <unknown>
ipykernel                 6.5.1            py38h595d716_0    <unknown>
ipython                   7.30.1           py38haa244fe_0    <unknown>
ipython_genutils          0.2.0                      py_1    <unknown>
jbig                      2.1               h8d14728_2003    <unknown>
jedi                      0.18.1           py38haa244fe_0    <unknown>
joblib                    1.1.0              pyhd8ed1ab_0    <unknown>
jpeg                      9d                   h8ffe710_0    <unknown>
jupyter_client            7.1.0              pyhd8ed1ab_0    <unknown>
jupyter_core              4.9.1            py38haa244fe_1    <unknown>
keyring                   23.4.0           py38haa244fe_0    <unknown>
keyrings.alt              4.0.2              pyhd3deb0d_0    <unknown>
kiwisolver                1.3.2            py38hbd9d945_1    <unknown>
lcms2                     2.12                 h2a16943_0    <unknown>
lerc                      3.0                  h0e60522_0    <unknown>
libblas                   3.9.0           12_win64_openblas    <unknown>
libbrotlicommon           1.0.9                h8ffe710_6    <unknown>
libbrotlidec              1.0.9                h8ffe710_6    <unknown>
libbrotlienc              1.0.9                h8ffe710_6    <unknown>
libcblas                  3.9.0           12_win64_openblas    <unknown>
libclang                  11.1.0          default_h5c34c98_1    <unknown>
libdeflate                1.8                  h8ffe710_0    <unknown>
libflang                  5.0.0           h6538335_20180525    <unknown>
liblapack                 3.9.0           12_win64_openblas    <unknown>
liblapacke                3.9.0           12_win64_openblas    <unknown>
libopenblas               0.3.18          pthreads_hc469a61_0    <unknown>
libpng                    1.6.37               h1d00b33_2    <unknown>
libsodium                 1.0.18               h8d14728_1    <unknown>
libtiff                   4.3.0                hd413186_2    <unknown>
libxgboost                1.5.0                h0e60522_1    <unknown>
libzlib                   1.2.11            h8ffe710_1013    <unknown>
llvm-meta                 5.0.0                         0    <unknown>
lmfit                     1.0.3                    pypi_0    pypi
lockfile                  0.12.2                     py_1    <unknown>
lz4-c                     1.9.3                h8ffe710_1    <unknown>
m2w64-gcc-libgfortran     5.3.0                         6    <unknown>
m2w64-gcc-libs            5.3.0                         7    <unknown>
m2w64-gcc-libs-core       5.3.0                         7    <unknown>
m2w64-gmp                 6.1.0                         2    <unknown>
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    <unknown>
matplotlib-base           3.5.0            py38h1f000d6_0    <unknown>
matplotlib-inline         0.1.3              pyhd8ed1ab_0    <unknown>
msgpack-python            1.0.3            py38hbd9d945_0    <unknown>
msys2-conda-epoch         20160418                      1    <unknown>
munkres                   1.1.4              pyh9f0ad1d_0    <unknown>
nest-asyncio              1.5.4              pyhd8ed1ab_0    <unknown>
networkx                  2.6.3              pyhd8ed1ab_1    <unknown>
numpy                     1.19.5           py38h09042cb_2    <unknown> (*This version was before the installation of the Add*)
olefile                   0.46               pyh9f0ad1d_1    <unknown>
openblas                  0.3.18          pthreads_h543f93c_0    <unknown>
openjpeg                  2.4.0                hb211442_1    <unknown>
openmp                    5.0.0                    vc14_1  [vc14]  <unknown>
openpyxl                  3.0.9              pyhd8ed1ab_0    <unknown>
openssl                   1.1.1l               h8ffe710_0    <unknown>
opentsne                  0.6.1            py38he5193b3_1    <unknown>
orange-canvas-core        0.1.23             pyhd8ed1ab_0    <unknown>
orange-spectroscopy       0.6.3                    pypi_0    pypi
orange-widget-base        4.15.1             pyhd8ed1ab_0    <unknown>
orange3                   3.30.2           py38h5d928e2_0    <unknown>
packaging                 21.3               pyhd8ed1ab_0    <unknown>
pandas                    1.3.4            py38h5d928e2_1    <unknown>
param                     1.12.0                   pypi_0    pypi
parso                     0.8.3              pyhd8ed1ab_0    <unknown>
pebble                    4.6.3                    pypi_0    pypi
pickleshare               0.7.5                   py_1003    <unknown>
pillow                    8.4.0            py38h794f750_0    <unknown>
pip                       21.3.1             pyhd8ed1ab_0    <unknown>
prompt-toolkit            3.0.22             pyha770c72_0    <unknown>
py-xgboost                1.5.0            py38haa244fe_1    <unknown>
pycparser                 2.21               pyhd8ed1ab_0    <unknown>
pyct                      0.4.8                    pypi_0    pypi
pygments                  2.10.0             pyhd8ed1ab_0    <unknown>
pyopenssl                 21.0.0             pyhd8ed1ab_0    <unknown>
pyparsing                 3.0.6              pyhd8ed1ab_0    <unknown>
pyqt                      5.12.3           py38haa244fe_8    <unknown>
pyqt-impl                 5.12.3           py38h885f38d_8    <unknown>
pyqt5-sip                 4.19.18          py38h885f38d_8    <unknown>
pyqtchart                 5.12             py38h885f38d_8    <unknown>
pyqtgraph                 0.12.2             pyhd8ed1ab_0    <unknown>
pyqtwebengine             5.12.1           py38h885f38d_8    <unknown>
pysocks                   1.7.1            py38haa244fe_4    <unknown>
python                    3.8.8           h7840368_0_cpython    <unknown>
python-dateutil           2.8.2              pyhd8ed1ab_0    <unknown>
python-louvain            0.15               pyhd3deb0d_0    <unknown>
python_abi                3.8                      2_cp38    <unknown>
pytz                      2021.3             pyhd8ed1ab_0    <unknown>
pywin32                   302              py38h294d835_2    <unknown>
pywin32-ctypes            0.2.0           py38haa244fe_1004    <unknown>
pyyaml                    6.0              py38h294d835_3    <unknown>
pyzmq                     22.3.0           py38h09162b1_1    <unknown>
qasync                    0.22.0             pyhd8ed1ab_0    <unknown>
qt                        5.12.9               h5909a2a_4    <unknown>
qtconsole                 5.2.1              pyhd8ed1ab_0    <unknown>
qtpy                      1.11.2             pyhd8ed1ab_0    <unknown>
renishawwire              0.1.15                   pypi_0    pypi
requests                  2.26.0             pyhd8ed1ab_1    <unknown>
rfc3986                   1.5.0              pyhd8ed1ab_0    <unknown>
scikit-learn              0.23.2           py38ha09990b_3    <unknown>
scipy                     1.5.3            py38h5f893b4_0    <unknown>
serverfiles               0.3.0                      py_0    <unknown>
setuptools                59.4.0           py38haa244fe_0    <unknown>
six                       1.16.0             pyh6c4a22f_0    <unknown>
sniffio                   1.2.0            py38haa244fe_2    <unknown>
spectral                  0.22.4                   pypi_0    pypi
sqlite                    3.37.0               h8ffe710_0    <unknown>
threadpoolctl             3.0.0              pyh8a188c0_0    <unknown>
tk                        8.6.11               h8ffe710_1    <unknown>
tornado                   6.1              py38h294d835_2    <unknown>
traitlets                 5.1.1              pyhd8ed1ab_0    <unknown>
typing_extensions         4.0.1              pyha770c72_0    <unknown>
ucrt                      10.0.20348.0         h57928b3_0    <unknown>
uncertainties             3.1.6                    pypi_0    pypi
unicodedata2              13.0.0.post2     py38h294d835_4    <unknown>
urllib3                   1.26.7             pyhd8ed1ab_0    <unknown>
vc                        14.2                 hb210afc_5    <unknown>
vs2015_runtime            14.29.30037          h902a5da_5    <unknown>
wcwidth                   0.2.5              pyh9f0ad1d_2    <unknown>
wheel                     0.37.0             pyhd8ed1ab_1    <unknown>
win_inet_pton             1.1.0            py38haa244fe_3    <unknown>
xgboost                   1.5.0            py38haa244fe_1    <unknown>
xlrd                      2.0.1              pyhd8ed1ab_3    <unknown>
xlsxwriter                3.0.2              pyhd8ed1ab_0    <unknown>
xz                        5.2.5                h62dcd97_1    <unknown>
yaml                      0.2.5                he774522_0    <unknown>
zeromq                    4.3.4                h0e60522_1    <unknown>
zipp                      3.6.0              pyhd8ed1ab_0    <unknown>
zlib                      1.2.11            h8ffe710_1013    <unknown>
zstd                      1.5.0                h6255e5f_0    <unknown>

Conda --ver : conda 4.9.2

**Comment by markotoplak** [comment]
@echavezangel, thanks! Reopening because it needs to be seen why (or if) installing add-ons on Windows upgrades numpy. If it does, we also need new Windows packages.

**Comment by markotoplak** [comment]
I tried 3 add-ons on a test Windows computer. I confirm, `Text` really forced the numpy upgrade. Interesting, it seems that one of its dependencies requires numpy newer that 1.19, at least on conda-forge.

To somewhat fix this issue or at least make Orange users suffer from it less often:
- We need to make new packages (almost done, waiting for MacOS signatures).
- The new release of Orange-Text should require Orange 3.31.1.

**Comment by PrimozGodec** [comment]
Now the new release of Orange3-text is made. Updating it to the newest version 1.6.2 should also fix Orange since it updates it.
